src/ccache.cpp

```diff
@@ -2207,6 +2209,9 @@ finalize_stats_and_trigger_cleanup(Context& ctx)
     if (result) {
       LOG("Result: {}", *result);
     }
+    if (Logging::results()) {
```

Comment:
The `!config.log_file().empty() || config.debug()` condition above makes stats logging enabled only if a log file or debug is specified. Is that intentional? I would expect stats logging to not depend on those configuration options.

In fact, how about not putting stats log code in Logging at all since it isn't really related?

Here's a sketch on how I think it could look like:

```c++
if (result && !config.stats_log.empty()) {
  const auto result_id = Statistics::get_result_id(); // new function: optional<string> Statistics::get_result_id();
  ASSERT(result_id);
  File stats_log(config.stats_log(), "a");
  PRINT(*stats_log, "# {}\n", ctx.args_info.input_file);
  PRINT(*stats_log, "{}\n", *result_id);
}
```

Reply:
The stats log was supposed to be small enough to run _without_ debugging, so that was not intentional.